pkg/apis/triggers/v1alpha1/trigger_template_types.go

```diff
@@ -18,12 +18,26 @@ package v1alpha1
 
 import (
 	"encoding/json"
+
 	pipelinev1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"github.com/tidwall/gjson"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/apis"
 )
 
+// apiVersion for Tekton core types
+const tektonApiVersion = GroupName + "/v1alpha1"
+
+// allowedTemplate types are the resource types identified by apiVersion + kind
+// that can be templated using TriggerResourceTemplates
+// TODO: Make this configurable using a ConfigMap entry
+var allowedTemplateTypes = map[string][]string{
+	tektonApiVersion: {"pipelineresource", "pipelinerun", "taskrun", "pipeline", "task", "condition"},
```

Comment:
There is no permission to create Triggers resources, which might be okay, but this is one reason the e2e tests are currently failing as we have a `TriggerTemplate` creating another template. Also note that creating a `TriggerTemplate` within a `TriggerTemplate` with the below validation (at least one resource) is a recursive problem. Also, the kinds within this value are not capitalized appropriately (the other reason validation is failing). We could use the strict capitalization or use `ToLower()`.

Reply:
used strings.ToLower() :man_facepalming:

So, the allowedKinds do not include TriggerTemplates at the moment which I guess is one way of avoiding the recursion problem